test/test-remote/testutils/index.ts

```diff
@@ -905,25 +907,43 @@ export async function waitForQueryResult<T>(
   // Callback for executing the query itself
   queryFunc: { (): Promise<T> },
   // Callback for checking the result for an expected value, returning null if the query should retry
-  testFunc: { (queryResponse: T): any | null},
+  testFunc: { (queryResponse: T): any | null },
   maxWaitSeconds = 30,
   logQueryResponse = false
 ) {
   const deadline = mtimeDeadlineInSeconds(maxWaitSeconds);
-  log.info("Waiting for query to return a match, deadline in %ss", maxWaitSeconds);
+  log.info(
+    "Waiting for query to return a match, deadline in %ss",
+    maxWaitSeconds
+  );
 
   while (true) {
     if (mtime() > deadline) {
       log.error("Reached %ss deadline, giving up", maxWaitSeconds);
       break;
     }
 
-    const data = await queryFunc();
+    let data: any;
+    try {
+      data = await queryFunc();
```

Comment:
CC @nickbp 

So, `queryFunc()` is expected to throw errors of type `got.RequestError` upon transient issues such as DNS resolution or timeout errors. I have added this to the docstring. It's an important property of this func all other functions that carry a `got....()` HTTP client call that is _not_ wrapped in a corresponding err handler.

Here, in this query loop, it of course makes sense to not crash the loop when this happens, but to go into the next loop iteration. 

What's important:
- the log message, so this does not happen silently
- to only catch errors emitted by the HTTP client, not other errors such as programming errors -- these are re-thrown.